src/app-layout/visual-refresh/drawers.tsx

```diff
@@ -169,6 +236,13 @@ function DesktopTriggers() {
     return null;
   }
 
+  const overflowItemHasBadge = () => {
+    const { overflowItems } = splitItems(drawers?.items);
+    return (overflowItems && overflowItems.filter(item => item.badge).length > 0) || false;
+  };
+
+  const { visibleItems, overflowItems } = splitItems(getDrawerItems());
+
```

Comment:
A bit of personal opinion, but I would simplify this code

```js
const { visibleItems, overflowItems } = splitItems(getDrawerItems());
const overflowItemHasBadge = overflowItems.find(item => item.badge);
```

No need for a function and also we can call `splitItems` only once